src/idl_parser.cpp

```diff
@@ -2175,8 +2163,16 @@ CheckedError Parser::ParseEnum(const bool is_union, EnumDef **dest) {
     }
   }
 
-  if (false == strict_ascending)
-    enum_def->SortByValue();  // Must be sorted to use MinValue/MaxValue.
+  enum_def->SortByValue();  // Must be sorted to use MinValue/MaxValue.
+
+  // Ensure enum value uniqueness.
+  auto prev_it = enum_def->Vals().begin();
+  for (auto it = prev_it + 1; it != enum_def->Vals().end(); ++it) {
+    auto prev_ev = *prev_it;
+    auto ev = *it;
+    if (prev_ev->GetAsUInt64() == ev->GetAsUInt64())
+      return Error("all enum values must be unique");
```

Comment:
Is it possible to clarify the error message by adding the value and names of clashed enumerators?

Reply:
Done. 

As an example, when I run it against the Direction enum I mentioned in the first comment, it will give an error message like this:

error : C:\path\to\schema\Direction.fbs(23, 1): error: all enum values must be unique: None and Center are both 0